app/src/main/java/me/saket/dank/ui/subreddit/SubredditActivity.java

```diff
@@ -491,6 +506,7 @@ private void setupSubmissionRecyclerView(@Nullable Bundle savedState) {
     subredditChangesStream
         .observeOn(mainThread())
         .takeUntil(lifecycle().onDestroy())
+        .doAfterNext(subreddit -> loadLastSortingAndPeriod(subreddit))
```

Comment:
This looks weird. This Rx chain is supposed to consume changes to the subreddit, but if I understand correctly, your intention is to start with the last used sorting right? 

You probably want to call `loadLastSortingAndPeriod()` in line `#240` instead. Something along the lines of:

```java
if (savedState != null && savedState.containsKey(KEY_SORTING_AND_TIME_PERIOD)) {
  //noinspection ConstantConditions
  sortingChangesStream.accept(savedState.getParcelable(KEY_SORTING_AND_TIME_PERIOD));
} else {
  SortingAndTimePeriod lastUsed = loadLastSortingAndPeriod();
  if (lastUsed != null) {
    sortingChangesStream.accept(lastUsed);
  } else {
    sortingChangesStream.accept(SortingAndTimePeriod.create(Reddit.Companion.getDEFAULT_SUBREDDIT_SORT()));
  }
}
```